src/App.tsx

```diff
@@ -15,47 +15,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.LENGTH:
+      visibleGoods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    case SortType.ALPABET:
+      visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    default:
+      break;
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setReversed] = useState(false);
+
+  const sortAlphabet = () => setSortType(SortType.ALPABET);
+
+  const sortByLength = () => setSortType(SortType.LENGTH);
+
+  const getReverse = () => setReversed(!isReversed);
+
+  const getReset = () => {
+    setSortType(SortType.NONE);
+    setReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={(
+            sortType === SortType.ALPABET
+              ? 'button is-info'
+              : 'button is-info is-light'
+          )}
+          onClick={sortAlphabet}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={(
+            sortType === SortType.LENGTH
+              ? 'button is-success'
+              : 'button is-success is-light'
+          )}
```

Comment:
Better install [classnames](https://www.npmjs.com/package/classnames) and use it instead of ternary operator.
The ternary operator is very limited in its functionality to be used in classNames. (this way usually did in real projects and would be great to work with it now).